tests/unit/go/task.yaml

```diff
@@ -25,13 +29,13 @@ execute: |
         skip="${skip:-} SKIP_NAKEDRET=1"
     fi
 
-    su -l -c "cd /tmp/static-unit-tests/src/github.com/snapcore/snapd && \
+    tests.session -u test exec sh -c "cd /tmp/static-unit-tests/src/github.com/snapcore/snapd && \
```

Comment:
I think we should have a switch here for trusty, to use `su -l -c` on trusty so that we can still get unit test coverage on trusty, but all the rest of the systems which support `tests.session` can benefit from not leaking the processes as you noticed